llvm/lib/SYCLLowerIR/ModuleSplitter.cpp

```diff
@@ -297,6 +298,19 @@ void collectFunctionsAndGlobalVariablesToExtract(
   }
 }
 
+// Checks for use of undefined user functions and emits a warning message.
+void checkForCallsToUndefinedFunctions(const Module &M) {
+  if (AllowDeviceImageDependencies)
+    return;
+  for (const Function &F : M) {
+    if (!F.isIntrinsic() && !F.getName().starts_with("__") &&
```

Comment:
can we make this logic a helper somewhere? probably we have a few places that want to know if something is a user function or not

Reply:
I can pull this into a separate function. Not sure where it will reside though. may be local to this file?

Thanks

Reply:
sure, but lets also check if it can be reused anywhere in the same file

Reply:
Addressed in 3d3a31fb05851342a479e17ddbdfa06552bc7d18

Thanks